### DISCUSSION TITLE
Providers for LoggingEvents - Sequence

### DISCUSSION THREAD

**Discussion by alvhartanto** [logfellow/logstash-logback-encoder#488] [Q&A] [answered]
Hi, does sequence option have reset feature for every new trace id (e.g. spring sleuth trace id)?

If not, does sequence option have maximum values?

Thank you in advance

**Comment by philsttr** [comment, marked as answer]
> does sequence option have reset feature for every new trace id (e.g. spring sleuth trace id)

No, the log sequence id is unrelated to trace id / span id.

The trace id / span id are usually put into the MDC, and included in the json output along with other values in the MDC via the `mdc` provider.

> does sequence option have maximum values?

The maximum sequence id is `Long.MAX_VALUE`, which is `2^63 - 1`, which is `9,223,372,036,854,775,807`

**Comment by alvhartanto** [comment]
Hi @philsttr , thank you for the prompt answers, I got several questions again regarding sequence if you don't mind

1. So, the only way to reset the sequence is restarting the service itself?
2. What happened when sequence value reached `Long.MAX_VALUE`? Does it make the application crash?

Thank you in advance :)

**Comment by philsttr** [comment]
1. yes
2. It will roll over to `Long.MIN_VALUE`, but keep in mind that if the application logs a message every _micro_ second, it will take close to 300 thousand years before it rolls over

**Comment by alvhartanto** [comment]
Hi @philsttr , i see, okay then, thank you very much for the answers 👍
